sympy/sets/sets.py

```diff
@@ -1465,12 +1465,32 @@ def reduce(args):
                 unk = [x for x in s
                        if any(other.contains(x) not in (True, False) for other in other_args)]
                 if unk:
+                    for val in other_args:
+                        if val.is_FiniteSet:
+                            # all values from val which are contained in `s`
+                            # and are non-numbers
+                            t = [x for x in s if val.contains(x) is true and not x.is_number]
```

Comment:
If understand the question correctly then:
There is no difference between them `.contains` return `SymPy bool`, and using `in` for `True, False(python bool)` does not make a difference. Like
```
>>> FiniteSet(1, 2, 3).contains(1) in (True, False)  # testing for python bool
True
>>> FiniteSet(1, 2, 3).contains(1) in (S.true, S.false)   # testing for SymPy bool
True
```
So `in` operator checks `==` instead not with `is`.
```

I will look into the failing test. Perhaps i am sorry for not responding quickly.